app/questionnaire/placeholder_parser.py

```diff
@@ -178,7 +178,7 @@ def _resolve_value_source_list(
 
     def _get_routing_path_block_ids(
         self, data: Mapping | Sequence, sections_to_ignore: list | None = None
```

Comment:
Since we are here we could add some top level arguments to these generic types as well.

Reply:
Done

Reply:
I've added the top level args to some more generic types within the module as well, in most cases, looking at validator, it was instances of dictionaries with a possible combination of string, list or more dictionaries as values, so it seemed suitable to use Mapping[str, Any] as other places have done, but can revisit if not

Reply:
Should be fine, since we do this in other places.